suitesparse.sh

```diff
@@ -1,14 +1,68 @@
+#!/bin/bash
 
+# parse SuiteSparse version from first argument, a git tag that ends in the version (no leading v)
 if [[ $1 =~ refs/tags/([0-9]*\.[0-9]*\.[0-9]*)\..*$ ]];
 then
     VERSION=${BASH_REMATCH[1]}
 else
+    echo "Specify a SuiteSparse version, such as: $0 refs/tags/7.4.3.0"
     exit -1
 fi
 echo VERSION: $VERSION
 
+NPROC="$(nproc)"
+if [ -z "${NPROC}" ]; then
+    # Default for platforms that don't have nproc. Mostly Windows.
+    NPROC="2"
+fi
+
+cmake_params=()
+if [ -n "${OpenMP_C_FLAGS}" ]; then
+    # macOS OpenMP flags.
+    # wheels.yml sets them as env vars, but CMake needs them as -D defs.
+    cmake_params+=(-DOpenMP_C_FLAGS="${OpenMP_C_FLAGS}")
+    cmake_params+=(-DOpenMP_C_LIB_NAMES="${OpenMP_C_LIB_NAMES}")
+    cmake_params+=(-DOpenMP_libomp_LIBRARY="${OpenMP_libomp_LIBRARY}")
+fi
+
+if [ -n "${CMAKE_GNUtoMS}" ]; then
+    # Windows needs .lib libraries, not .a
+    cmake_params+=(-DCMAKE_GNUtoMS=ON)
+    # Windows expects 'graphblas.lib', not 'libgraphblas.lib'
+    cmake_params+=(-DCMAKE_SHARED_LIBRARY_PREFIX=)
+    cmake_params+=(-DCMAKE_STATIC_LIBRARY_PREFIX=)
+fi
+
 curl -L https://github.com/DrTimothyAldenDavis/GraphBLAS/archive/refs/tags/v${VERSION}.tar.gz | tar xzf -
 cd GraphBLAS-${VERSION}/build
-cmake .. -DCMAKE_BUILD_TYPE=Release
-make -j$(nproc)
+
+# Disable optimizing some rarely-used types for significantly faster builds and significantly smaller wheel size.
```

Comment:
If experimenting with the workflow I recommend disabling *all* the types. That brings the GraphBLAS build times into the 15 minute range on GitHub Actions.

Reply:
I have a non-documented feature that you can use for this.  Turn on the CMAKE_CUDA_DEV flag in the GraphBLAS CmakeLists.txt.  That disables all compilation of the kernels in Source/Generated2.

I don't recommend that for production so I don't document it.  The resulting GraphBLAS will be pretty slow.  But it's nice for development.

Reply:
@DrTimothyAldenDavis I thought I was going crazy because `-DCMAKE_CUDA_DEV=1` didn't seem to make a difference, but I just noticed CMakeLIsts.txt has `set ( CMAKE_CUDA_DEV off )` on lines 56 and 61, so `CMAKE_CUDA_DEV` is only usable if one is able to edit CMakeLists.txt directly. I supoose it's possible to do with `sed`, but `sed` inplace support varies between OSes and such edits are a PITA.

Reply:
@DrTimothyAldenDavis Just FYI, compiling that combinatorial explosion of types really does make working with GraphBLAS very difficult. This PR would not happen if GitHub's accounting wasn't broken, as the CI time should have cost me several hundred dollars. With the types disabled as seen here, building just the macOS wheels one time costs about $3.50 according to GitHub's published price list ($5 for all OSes). I don't mind because GitHub doesn't have my credit card and they aren't counting anything against my free quota, but that's just luck on our part.

Reply:
That's intentional.  It is not meant for anyone else to use but me.  It is supposed be very hard to turn on GBCUDA_DEV, because turning it on slows down GraphBLAS immensely (run time, not compile time).

If it's easy to turn on GBCUDA_DEV, then I worry that it would be compiled that way for production and distribution, as in a linux distro or conda package.  That would be awful.

If you like, you could instead compile with 

`-DGxB_NO_BOOL=1 -DGxB_NO_INT8=1 -DGxB_NO_INT16=1` ...  (etc).

I do not recommend that for a production build.

See [GraphBLAS/Source/GB_control.h:](https://github.com/DrTimothyAldenDavis/GraphBLAS/blob/stable/Source/GB_control.h)
Note that I already disable quite a few semirings via that file.

GraphBLAS requires a surprisingly large amount of semirings and operators when used in production.

I'm working on a JIT where I can compile these at run-time, however.  That will keep the performance up (and even improve it) while at the same time reducing the initial compile time.  Distros will have to package my source code for that to work, however.